.ci/test_r_package_windows.ps1

```diff
@@ -0,0 +1,110 @@
+# Download a file and retry upon failure. This looks like
+# an infinite loop but CI-level timeouts will kill it
+function Download-File-With-Retries {
+  param(
+    [string]$url,
+    [string]$destfile
+  )
+  do {
+    Write-Output "Downloading ${url}"
+    sleep 5;
+    (New-Object System.Net.WebClient).DownloadFile($url, $destfile)
+  } while(!$?);
+}
+
+$env:R_WINDOWS_VERSION = "3.6.3"
+$env:R_LIB_PATH = "$env:BUILD_SOURCESDIRECTORY/RLibrary" -replace '[\\]', '/'
+Write-Output "R_LIB_PATH: $env:R_LIB_PATH"
+$env:PATH = "$env:R_LIB_PATH/Rtools/bin;" + "$env:R_LIB_PATH/R/bin/x64;" + "$env:R_LIB_PATH/miktex/texmfs/install/miktex/bin/x64;" + $env:PATH
+$env:CRAN_MIRROR = "https://cloud.r-project.org/"
+$env:CTAN_MIRROR = "https://ctan.math.illinois.edu/systems/win32/miktex/tm/packages/"
+
+if ($env:COMPILER -eq "MINGW") {
+  $env:CXX = "$env:R_LIB_PATH/Rtools/mingw_64/bin/g++.exe"
+  $env:CC = "$env:R_LIB_PATH/Rtools/mingw_64/bin/gcc.exe"
+}
+
+cd $env:BUILD_SOURCESDIRECTORY
+tzutil /s "GMT Standard Time"
+[Void][System.IO.Directory]::CreateDirectory($env:R_LIB_PATH)
+
+if ($env:COMPILER -eq "MINGW") {
+  Write-Output "Telling R to use MinGW"
+  $install_libs = "$env:BUILD_SOURCESDIRECTORY/R-package/src/install.libs.R"
+  ((Get-Content -path $install_libs -Raw) -replace 'use_mingw <- FALSE','use_mingw <- TRUE') | Set-Content -Path $install_libs
+}
+
+# download R and RTools
+Write-Output "Downloading R and Rtools"
+Download-File-With-Retries -url "https://cloud.r-project.org/bin/windows/base/old/$env:R_WINDOWS_VERSION/R-$env:R_WINDOWS_VERSION-win.exe" -destfile "R-win.exe"
+Download-File-With-Retries -url "https://cloud.r-project.org/bin/windows/Rtools/Rtools35.exe" -destfile "Rtools.exe"
+
+# Install R
+Write-Output "Installing R"
+Start-Process -FilePath R-win.exe -NoNewWindow -Wait -ArgumentList "/VERYSILENT /DIR=$env:R_LIB_PATH/R /COMPONENTS=main,x64" ; Check-Output $?
+Write-Output "Done installing R"
+
+Write-Output "Installing Rtools"
+Start-Process -FilePath Rtools.exe -NoNewWindow -Wait -ArgumentList "/VERYSILENT /DIR=$env:R_LIB_PATH/Rtools" ; Check-Output $?
+Write-Output "Done installing Rtools"
+
+# MiKTeX and pandoc can be skipped on non-MINGW builds, since we don't
+# build the package documentation for those
+if ($env:COMPILER -eq "MINGW") {
+
```

Comment:
Please remove this empty line.

Reply:
removed  in https://github.com/microsoft/LightGBM/pull/2936/commits/747378eb41b74d5124b905f9cca5f76727d3a5ae